packages/core/data-transfer/src/utils/json.ts

```diff
@@ -38,13 +38,23 @@ export const diff = (a: unknown, b: unknown, ctx: Context = createContext()): Di
     return diffs;
   };
 
-  if (aType === 'undefined') {
-    return added();
-  }
+  const dataType = () => {
+    diffs.push({
+      kind: 'dataType',
+      path,
+      types: [aType, bType],
+      values: [a, b],
+    });
+    return diffs;
+  };
 
-  if (bType === 'undefined') {
-    return deleted();
-  }
+  const isLooselyEqual = () => {
```

Comment:
I did this because originally I had checks for aType and bType that didn't seem necessary, but I left it because it feels better to have our own function instead of directly using ==.  If anyone disagrees I can remove this and just use == down below.